include/ros1_bridge/factory.hpp

```diff
@@ -152,6 +152,92 @@ class Factory : public FactoryInterface
     ROS1_T & ros1_msg);
 };
 
+template <class R1_TYPE, class R1_REQ, class R1_RES, class R2_TYPE, class R2_REQ, class R2_RES>
+class ServiceFactory : public ServiceFactoryInterface
+{
+private:
+
+  void translate_1_to_2(const R1_REQ&, R2_REQ&);
+  void translate_1_to_2(const R1_RES&, R2_RES&);
```

Comment:
Yes, currently the implementation of these functions iterate over the fields. I think we can get rid of this of we expand the already existing message templates for the request / response. I will look into it...